### DISCUSSION TITLE
If it is possible to calculate splitChunks minSize after tree shaking and minimized?

### DISCUSSION THREAD

**Discussion by lnwu** [webpack/webpack#14263] [Q&A]
## Feature request

For now seems like the splitChunks minSize calculated raw file size， I have a [utils.ts](https://gist.github.com/lnwu/a08f551659a77d2f26036afee51fa224#file-utils-ts), the file size more than 5kb with comments, otherwise not, the [config](https://gist.github.com/lnwu/a08f551659a77d2f26036afee51fa224#file-webpack-config-js-L17) for min size is 5000, the chunk is split.

With comments (the chunk less 1kb after minimize but still split)
![image](https://user-images.githubusercontent.com/16202655/133575251-9ee23dba-cc99-44fd-afd1-1918d13182c3.png)

After remove comments in [utils.ts](https://gist.github.com/lnwu/a08f551659a77d2f26036afee51fa224#file-utils-ts), the chunk is removed.
![image](https://user-images.githubusercontent.com/16202655/133574966-d6b014a3-07f7-4217-acd1-ea5e6c871f8b.png)

**What is the expected behavior?**

minSize can be calculated after minimize and tree shaking

**What is motivation or use case for adding/changing the behavior?**
it does make sense if we set min size 5kb chunks, and the chunks will more than 5kb. 😄   
max size as well, for now, default maxSize 0, the node_modules chunks is 180kb, after I set maxSize to 100kb, I think there are two chunks, but there is three, and left two chunks total size is less than 100kb, I m confused why it can not be one chunk.
![image](https://user-images.githubusercontent.com/16202655/133578605-7eb9475e-437f-402e-aa72-90332caeb33a.png)

**How should this be implemented in your opinion?**
Add a flag for now and set the default to true for the next major version

**Are you willing to work on this yourself?**
yes

**Comment by alexander-akait** [comment]
No, it is impossible

**Reply by lnwu** [reply]
Is it technically impossible?
If it makes sense I may try to figure out if it can be implemented.

**Reply by alexander-akait** [reply]
we will minimize after code generation

**Reply by lnwu** [reply]
I mean it is possible to minimize first and then do code generation

**Reply by alexander-akait** [reply]
No, it is impossible
